### PR TITLE
SI-5942 toolboxes now reset front ends

### PR DESCRIPTION
**Pull request by xeno-by**
FrontEnd => Reporter proxy now correctly redirects
flush and reset back to the underlying front end.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @paulp @retronym @gkossakowski

File: test/files/run/t5942.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import scala.reflect.runtime.universe._
+import scala.reflect.runtime.{currentMirror => cm}
+import scala.tools.reflect._
+
+object Test extends App {
+  val tb = cm.mkToolBox()
+  tb.parse("def x = {}")
+  try { tb.parse("def x = {") } catch { case _ => }
+  tb.parse("def x = {}")
```
Comment (gkossakowski) [inline test/files/run/t5942.scala:9]: Before applying your fix this line would fail with an error?

Comment (xeno-by) [inline test/files/run/t5942.scala:9]: That's correct.

**Comment by gkossakowski** [comment]
LGTM.

**Pull request merged**